File: 백준/Silver/11055.가장 큰 증가 부분 수열/가장 큰 증가 부분 수열.java
import java.util.*;
import java.io.*;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        int[] dp = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());

        for(int i = 0; i < n; i++){
            arr[i] = Integer.parseInt(st.nextToken());
            dp[i] = arr[i];
        }

        for(int i = 1; i < n;i++){
            for(int j = 0; j < i; j++){
                if(arr[i] > arr[j]){
                    dp[i] = Math.max(dp[j] + arr[i], dp[i]);
                }
            }
            
        }

        int result = Arrays.stream(dp).max().getAsInt();
        System.out.println(result);

    }
}

File: 백준/Silver/11055. 가장 큰 증가 부분 수열/가장 큰 증가 부분 수열.py
import sys

n = int(sys.stdin.readline())

arr = list(map(int, sys.stdin.readline().split()))

dp = [0] * n

for i in range(n):
    dp[i] = arr[i]

for i in range(1, n):
   for j in range(i):
       if arr[j] < arr[i]:
           dp[i] = max(dp[i], dp[j] + arr[i])

print(max(dp))